libs/state/src/lib/convert-partials-to.observable.ts

```diff
@@ -0,0 +1,45 @@
+import {
+  combineLatest,
+  isObservable,
+  Observable,
+  startWith,
+  timer,
+} from 'rxjs';
+import { Injector, isSignal, Signal } from '@angular/core';
+import { toObservable } from '@angular/core/rxjs-interop';
+import { map } from 'rxjs/operators';
+import { coalesceWith } from '@rx-angular/cdk/coalescing';
+
+export function convertPartialsToObservable<
+  T extends object,
+  K extends keyof T
+>(
+  input: Partial<{ [K: string]: Observable<T[K]> | Signal<T[K]> }>,
+  injector?: Injector
+): Observable<Partial<T>> {
+  const keys = Object.keys(input);
+
+  // eslint-disable-next-line @typescript-eslint/no-unsafe-return
+  const observables = keys.map((key) => {
+    const obsOrSignal = input[key];
+
+    if (isObservable(obsOrSignal)) {
+      return obsOrSignal?.pipe(startWith(undefined));
+    }
+    if (isSignal(obsOrSignal)) {
+      return toObservable(obsOrSignal, { injector });
+    }
+  });
+
+  return combineLatest(observables).pipe(
```

Comment:
should I filter out any key which has a value of undefined?
Undefined might come from line 27